internal/provider/resources/deployment.go

```diff
@@ -315,15 +319,13 @@ func (r *DeploymentResource) Schema(_ context.Context, _ resource.SchemaRequest,
 			"concurrency_limit": schema.Int64Attribute{
 				Description: "The deployment's concurrency limit.",
 				Optional:    true,
-				Computed:    true,
 				Validators: []validator.Int64{
 					int64validator.AtLeast(1),
 				},
 			},
 			"concurrency_options": schema.SingleNestedAttribute{
 				Description: "Concurrency options for the deployment.",
 				Optional:    true,
-				Computed:    true,
```

Comment:
Whew, weird. I wonder if we need to be making more use of `.IsUnknown()` in the resource logic (a very unfinished thought)

Reply:
one more thing to add - two ways to avoid `unknown` values during plan time:

1. Optional=true, Computed=true + add a Default value

> // Default defines a proposed new state (plan) value for the attribute
	// if the configuration value is null. Default prevents the framework
	// from automatically marking the value as unknown during planning when
	// other proposed new state changes are detected. If the attribute is
	// computed and the value could be altered by other changes then a default
	// should be avoided and a plan modifier should be used instead.

OR

2. Optional=true (only) -- this makes an omitted value to be `null`

Reply:
> I wonder if we need to be making more use of .IsUnknown() in the resource logic (a very unfinished thought)

i think we definitely should, same with `.IsNull()`.  That said, my gut is that 90% of the time we won't need to, because null values should be treated as null API attributes in the network call

for IsUnknown, im thinking we'll use it more after this PR, since my hope is that we have a slightly better understanding of it